framework/arcane-framework/src/main/scala/services/storage/models/amazon/AmazonS3StoragePath.scala

```diff
@@ -0,0 +1,62 @@
+package com.sneaksanddata.arcane.framework
+package services.storage.models.amazon
+
+import services.storage.models.base.BlobPath
+
+import scala.util.matching.Regex
+import scala.util.{Failure, Success, Try}
+
+/**
+  * Represents a path to a blob in Amazon S3 storage.
+  *
+  * @param bucket The name of the bucket.
+  * @param objectKey The key of the object in the bucket.
+  */
+private final case class AmazonS3StoragePath(bucket: String, objectKey: String) extends BlobPath {
+
+  /**
+   * Converts the path to a HDFS-style path.
+   *
+   * @return The path as a string.
+   */
+  override def toHdfsPath = s"s3a://$bucket/$objectKey"
+
+  /**
+    * Joins the given key name to the current path.
+    *
+    * @param keyName The key name to join.
+    * @return The new path.
+    */
+  def join(keyName: String) = new AmazonS3StoragePath(bucket, if (objectKey.isEmpty) keyName else s"$objectKey/$keyName")
+
+}
+
+/**
+  * Companion object for [[AmazonS3StoragePath]].
+  */
+object AmazonS3StoragePath {
+  private val matchRegex: String = "s3a://([^/]+)/?(.*)"
+
+  /**
+    * Checks if the given path is an Amazon S3 path.
+    *
+    * @param hdfsPath The path to check.
+    * @return True if the path is an Amazon S3 path, false otherwise.
+    */
+  def isAmazonS3Path(hdfsPath: String): Boolean = hdfsPath.matches(matchRegex)
+
+  /**
+    * Creates an [[AmazonS3StoragePath]] from the given HDFS path.
+    *
+    * @param hdfsPath The HDFS path.
+    * @return The [[AmazonS3StoragePath]].
+    */
+  def fromHdfsPath(hdfsPath: String): Try[AmazonS3StoragePath] = {
```

Comment:
you need to define `apply` for the method to be a valid monad. I think this method looks like a good candidate? 
it acts as a default constructor:
val myPath = AmazonS3StoragePath("s3a://...")
You can also provide overloads for `apply` AND apply quality checks in it - like the one you do in `isAmazonS3Path`

Reply:
Fixed